runtime/onert/core/src/compiler/ExecutorFactory.cc

```diff
@@ -96,31 +97,115 @@ void initializeSubgraphIOTensors(compiler::LoweredGraph &lowered_graph,
+      auto new_operand_ind = partial_graph.addOperand(operand_ind, std::move(new_operand));
+      UNUSED_RELEASE(new_operand_ind);
+      assert(new_operand_ind == operand_ind);
+    });
+  // Separate operations into partial graphs
+  whole_graph.operations().iterate(
+    [&](const ir::OperationIndex &op_ind, const ir::Operation &operation) {
+      auto &op_li = lgraph.lower_info().operation;
+      auto backend = op_li.at(op_ind).backend();
+      auto &partial_graph = *context_data_map[backend].graph;
+      auto &external_operands = context_data_map[backend].external_operands;
+      auto &operand_layouts = context_data_map[backend].operand_layouts;
+
+      {
+        // Add missing operands (externals)
+        auto io_list = (operation.getInputs() + operation.getOutputs()) | ir::Remove::DUPLICATED |
+                       ir::Remove::UNDEFINED;
+        for (auto operand_ind : io_list)
+        {
+          if (partial_graph.operands().exist(operand_ind))
+            continue;
+
+          // Copy the operand and insert it to the partial graph
+          const auto &operand = whole_graph.operands().at(operand_ind);
+          auto new_operand = std::make_unique<ir::Operand>(operand);
+          // TODO Introduce a method for resetting use/def values of Operand
+          // NOTE Use/Def info is going to be filled in `Graph::finishBuilding`
+          const_cast<ir::OperationIndexSet &>(new_operand->getUses()).clear();
+          new_operand->unsetDef();
+          auto new_operand_ind = partial_graph.addOperand(operand_ind, std::move(new_operand));
+          UNUSED_RELEASE(new_operand_ind);
+          assert(new_operand_ind == operand_ind);
+
+          auto layout =
+            lgraph.lower_info().operand.at(operand_ind).def_factors().getOnlyElement().layout();
+          assert(operand_layouts.find(operand_ind) == operand_layouts.end());
+          operand_layouts[operand_ind] = layout;
+          external_operands.add(operand_ind);
+        }
+
+        auto new_op_ind = partial_graph.addOperation(op_ind, clone(operation));
+        UNUSED_RELEASE(new_op_ind);
+        assert(new_op_ind == op_ind);
+      }
+    });
+
+  // Create contexts
+  auto whole_op_order = lgraph.graph().topolSortOperations();
+  for (auto &pair : context_data_map)
+  {
+    auto backend = pair.first;
+    auto &data = pair.second;
+    data.graph->finishBuilding();
+    // Handle graph input/outputs or external tensors
+    data.graph->operands().iterate([&](const ir::OperandIndex &ind, const ir::Operand &operand) {
+      if (whole_graph.getInputs().contains(ind) || whole_graph.getOutputs().contains(ind))
+        data.external_operands.add(ind);
+      // Inputs are either "graph input" or "no def op and non-constant"
+      if (whole_graph.getInputs().contains(ind) ||
+          (!operand.getDef().valid() && !operand.isConstant()))
+        // Outputs are either "graph output" or "no uses"
+        data.graph->addInput(ind);
```

Comment:
I have a question.
Are the Input/Output of all partial graphs the same as the Input/Output of the whole graph?

Reply:
The whole graph's input/outputs are always input/outputs in partial graphs.
Plus, I had a lot of thoughts about it and the conclusion was like below.
Here's a figure as an example.

![NewBackendContext (2)](https://user-images.githubusercontent.com/10582184/106846131-a23c2680-66ef-11eb-929a-2ccd5ddc80e8.png)

Partial Graph Input

- The whole graph's input (Backend Green %0) OR
- There is no def operation and not a constant (Backend Red %1, Backend Green %2, %3)

Partial Graph Output

- The whole graph's output (Backend Green %5) OR
- There is no use operations (Backend Red %3, Backend Green %1)

I'm not sure this is the best way to define partial graph's I/O 😓 but it works with our current compilation implementation for now.